### PR TITLE
HDDS-7889. Intermittent failure in TestOzoneSnapshotRestore

### PR DESCRIPTION
**Pull request by mladjan-gadzic**
## What changes were proposed in this pull request?
The issue was that `TestOzoneSnapshotRestore::keyCopy` method was not finished copying keys when assertion for return value of `TestOzoneSnapshotRestore::keyCount` method was executed. It is proposed to use `GenericTestUtils::waitFor` to wait for `TestOzoneSnapshotRestore::keyCount` to return expected number of keys. For more info check comment section under linked Jira ticket.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7889

## How was this patch tested?
* unit tests

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneSnapshotRestore.java [hunk 1]
```diff
@@ -332,7 +342,12 @@ public void testRestoreSnapshotDifferentBucketLayout(
...
 
-    int finalKeyCount = keyCount(buck2, keyPrefix);
-    Assertions.assertEquals(5, finalKeyCount);
+    GenericTestUtils.waitFor(() -> {
+      try {
+        return 5 == keyCount(buck2, keyPrefix);
+      } catch (IOException e) {
+        return false;
+      }
+    }, 1000, 10000);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestOzoneSnapshotRestore.java:351]: Can you please extract the duplicated code into a method `waitForKeyCount`?

**Review by adoroszlai — COMMENTED** [review]
Thanks @mladjan-gadzic for working on this.

Please run the test repeatedly in CI (see https://github.com/apache/ozone/commit/aaa6e8be3ccf6d47d3dd93750757c13e286fa374 for an example) and post the results.

**Comment by mladjan-gadzic** [comment]
@adoroszlai thanks for the review! 

> Please run the test repeatedly in CI (see https://github.com/apache/ozone/commit/aaa6e8be3ccf6d47d3dd93750757c13e286fa374 for an example) and post the results.

I've been running those kind of tests for a quite some time. Please check [https://github.com/mladjan-gadzic/ozone/actions/workflows/HDDS-7889.yml](https://github.com/mladjan-gadzic/ozone/actions/workflows/HDDS-7889.yml). There has not been a single failure since the last commit. Is that what you are referring to?

**Comment by adoroszlai** [comment]
> I've been running those kind of tests for a quite some time. Please check https://github.com/mladjan-gadzic/ozone/actions/workflows/HDDS-7889.yml. There has not been a single failure since the last commit. Is that what you are referring to?

Yes, great work, thanks for the link.

**Comment by mladjan-gadzic** [comment]
Thanks @adoroszlai for the review!

**Pull request merged**